### DISCUSSION TITLE
blog/metube-selfhosted-youtube-downloader-with-a-sleek-web-interface/

### DISCUSSION THREAD

**Discussion by giscus** [AkashRajpurohit/AkashRajpurohit#75] [Comments]
# blog/metube-selfhosted-youtube-downloader-with-a-sleek-web-interface/

MeTube is a web UI for youtube-dl/yt-dlp that allows you to download videos from YouTube and other platforms. It's perfect for archiving your favorite content or downloading videos for offline viewing.

https://akashrajpurohit.com/blog/metube-selfhosted-youtube-downloader-with-a-sleek-web-interface/?ref=rss

**Comment by leanderkretschmer** [comment]
The webui isn't reachable at all, I used the default compose structure found on this page and only changed the download path and nothing seems to work

**Comment by kenohbee** [comment]
Hi, thanks for your work! I've been using this for one month and found it very useful. May I ask, is it possible to input the URL via context menu in MacOSX? I know I could build a custom command in the right click context menu in OSX, but how to pass the URL to metube? I am using docker version of metube. Thanks in advance!

**Comment by dwolfuk40** [comment]
Are you able to add a seperate env variable for the sqlite db. Trying to set the audio and video to a network share results in this failing and no gui appearing. I don't want large videos stored on my docker instance but the sqlite issue on networked volumes is common so it seems like a good idea to have the audio and video locations seperate and the db with the docker bits.

**Reply by dwolfuk40** [reply]
For others with network shares i figured out with the env variables how to store on network share

Set the STATE_DIR to a dir on the docker server, then DOWNLOAD_DIR and TEMP_DIR is environment. Be sure to have a matching volume. Then everything works as it should

        environment:
          - DOWNLOAD_DIR=/video-downloads
          - STATE_DIR=/state
          - TEMP_DIR=/temp
        volumes:
            - ./temp:/temp:rw
            - ./state:/state:rw
            - ./downloads:/downloads:rw
            - {NETWORK SHARE}/video:/video-downloads:rw
            - {NETWORK SHARE}/audio:/audio-downloads:rw
